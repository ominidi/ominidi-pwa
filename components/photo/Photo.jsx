import React from 'react';
import PropTypes from 'prop-types';

/**
 * Photo render a single photo.
 *
 * @author Gabriele D'Arrigo <darrigo.g@gmail.com>
 * @param props
 * @returns {XML}
 * @constructor
 */
const Photo = ({
  id,
  src,
  alt,
  caption,
}) => (
  <div className="photos__item" data-id={id}>
    <article className="card">
      <figure className="card__picture">
        <img
          src={src}
          title={alt}
          alt={alt}
        />
      </figure>

      <footer className="card__footer">
        <h5 className="card__caption title-6">
          { caption }
        </h5>
      </footer>
    </article>
  </div>
);

Photo.propTypes = {
  id: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
  caption: PropTypes.string,
};

Photo.defaultProps = {
  alt: null,
  caption: null,
};

export default Photo;
