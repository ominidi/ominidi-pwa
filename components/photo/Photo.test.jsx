import React from 'react';
import renderer from 'react-test-renderer';
import Photo from './Photo';

const data = {
  id: '1',
  alt: 'An alt',
  caption: 'A caption',
  src: 'http://www.foo/img.jpg',
};

describe('Photo', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Photo {...data} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
