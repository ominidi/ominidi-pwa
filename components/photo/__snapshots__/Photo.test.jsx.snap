// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo should render correctly 1`] = `
<div
  className="photos__item"
  data-id="1"
>
  <article
    className="card"
  >
    <figure
      className="card__picture"
    >
      <img
        alt="An alt"
        src="http://www.foo/img.jpg"
        title="An alt"
      />
    </figure>
    <footer
      className="card__footer"
    >
      <h5
        className="card__caption title-6"
      >
        A caption
      </h5>
    </footer>
  </article>
</div>
`;
